screens/AnnouncementsList.jsx

```diff
@@ -0,0 +1,168 @@
+import locales from 'constants/locales';
+import { FAnnouncementCard } from 'components/Scoped/Announcement/Card/FAnnouncementCard';
+import { getUserAnnouncementsService } from 'services/announcement/getUserAnnouncements.service';
+
+export const AnnouncementsList = ({
+  isMe,
+  userId,
+  onlyActive = false,
+  onlyFavorites = false,
+  horizontal = false,
+  numColumns,
+  pullData,
+}) => {
+  const [
+    params,
+    setParams,
+  ] = useState({
+    pageSize: 8,
+    offset: 0,
+    onlyActive,
+    onlyFavorites,
+  });
+  const [
+    isLoading,
+    setIsLoading,
+  ] = useState(true);
+  const [
+    endReached,
+    setEndReached,
+  ] = useState(false);
+  const [
+    announcements,
+    setAnnouncements,
+  ] = useState([]);
+
+  const {
+    setShowErrorModal,
+    drawErrorModal,
+  } = useErrorModal();
+
+  useEffect(() => {
+    fetchAnnouncements();
+  }, [params]);
+
+  const fetchAnnouncements = async () => {
+    try {
+      if (isMe) {
+        const res = await getMyAnnouncementsService(params);
+        setAnnouncements([...announcements, ...res.data]);
+        if (res.data.length < 8) {
+          setEndReached(true);
+        }
+      }
+      if (userId) {
+        const res = await getUserAnnouncementsService(userId, params);
+        setAnnouncements([...announcements, ...res.data]);
+        pullData(announcements);
+        if (res.data.length < 8) {
+          setEndReached(true);
+        }
+      }
+    } catch {
+      setShowErrorModal(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleEnd = () => {
+    setParams(
+      (prevState) => ({
+        pageSize: prevState.pageSize + 8,
+        offset: prevState.offset + 8,
+        onlyActive: prevState.onlyActive,
+        onlyFavorites: prevState.onlyFavorites,
+      }),
+    );
+  };
+
+  const drawAnnouncementCard = ({ item }) => (
+    <View style={{ flexBasis: '50%' }}>
+      <FAnnouncementCard
+        width={horizontal ? sizes.WIDTH_180 : sizes.WIDTH_FULL}
+        height={280}
+        data={item}
+      />
+    </View>
+  );
+  const drawNoAnnouncementInfo = () => {
+    if (isLoading === false && announcements.length === 0) {
+      return (
+        <View style={horizontal ? styles.containerHorizontal : styles.containerVertical}>
+          <FHeading
+            align={placements.CENTER}
+            size={fonts.HEADING_SMALL}
+            color={colors.DARK_GRAY}
+            weight={fonts.HEADING_WEIGHT_REGULAR}
+            title={isMe ? locales.YOU_DONT_HAVE_ANY_ANNOUNCEMENTS_YET : locales.USER_DONT_HAVE_ANY_ANNOUNCEMENTS_YET}
+          />
+        </View>
+      );
+    }
+  };
+  const renderActivityIndicator = () => (!endReached
+    ? (
+      <ActivityIndicator
+        animating
+        size={Platform.OS === 'ios' ? 'large' : sizes.ICON_30}
+        color={colors.GRAY}
+        style={horizontal ? styles.containerHorizontal : ''}
+      />
+    ) : null);
+
+  return (
+    <>
+      {isLoading && (
```

Comment:
jezeli coś to coś, reszta nie powinna sie renderować. czyli dopóki nie mam ogłoszen to renderuje sie tylko spinner